### ISSUE TITLE
Add documentation about custom random DataPipe

### ISSUE THREAD

**Issue by NivekT** [pytorch/data#1105]
### 📚 The doc issue

Similar to #1104, `DataLoader2` has built-in functionality to set random seed for DataPipes when they have `set_seed` method implemented.

https://github.com/pytorch/data/blob/main/torchdata/dataloader2/graph/settings.py#L22

We should provide official guidance about what we expect users to implement in cases where a custom random DataPipe is implemented/necessary.
